DP Mystery Map/Assets/Scripts/DialogueManager.cs

```diff
@@ -0,0 +1,110 @@
+using Ink.Parsed;
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using Ink.Runtime;
+using PlayerInfo;
+using UnityEngine.UI;
+
+public class DialogueManager : MonoBehaviour
+{
+    [SerializeField] private GameObject dialogPanel;
+    
+    [SerializeField] private TextMeshProUGUI dialogText;
+
+    private Ink.Runtime.Story currentDialog;
+
+    public static bool dialogIsPlaying { get; set; }
+
+    private string fullText;
+    private float charRevealDelay = 0.05f;
+
+    public static DialogueManager instance { get; private set; }
```

Comment:
Add `OnDestroy()` and make it set `instance` to if `instance` is `this`. Otherwise, the DialogueManager can't be created after the first one is destroyed